In [1]:
import pandas as pd
data_frame = pd.read_csv("Glassdoor_jobs.csv")

In [2]:
data_frame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5578 entries, 0 to 5577
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Job Title          5578 non-null   object 
 1   Salary Estimate    5578 non-null   object 
 2   Job Description    5578 non-null   object 
 3   Rating             5578 non-null   float64
 4   Company Name       5578 non-null   object 
 5   Location           5578 non-null   object 
 6   Size               5578 non-null   object 
 7   Founded            5578 non-null   int64  
 8   Type of ownership  5578 non-null   object 
 9   Industry           5578 non-null   object 
 10  Sector             5578 non-null   object 
 11  Revenue            5578 non-null   object 
dtypes: float64(1), int64(1), object(10)
memory usage: 523.1+ KB


In [3]:
data_frame.head(50)

Job Title  \
0             Geospatial Researcher/Data Scientist IV   
1                               Senior Data Scientist   
2   Software Integrator / Data Scientist - Expert ...   
3                                 Lead Data Scientist   
4                                      Data Scientist   
5                                      Data Scientist   
6                           Defense Data Scientist II   
7                                 Senior Data Analyst   
8                                      Data Scientist   
9                     Defense Senior Data Scientist I   
10                                     Data Scientist   
11                                     Data Scientist   
12                 2023771 Data Scientist $215,000.00   
13                              DATA SCIENTIST - USMS   
14                                Junior Data Labeler   
15                                     Data Scientist   
16                     Tech Excellence Data Scientist   
17                                     Data Scientist   
18                              Senior Data Scientist   
19                                     Data Scientist   
20               Jr. Data Scientist (Remote Eligible)   
21                                   Data Scientist I   
22                                     Data Scientist   
23                              Senior Data Scientist   
24               Data Scientist - Electricity Markets   
25                Data Scientist - Clearance Required   
26                      Data Scientist with Polygraph   
27                                     Data Scientist   
28             Senior Data Scientist (Remote/On Site)   
29                                     Data Scientist   
30                         Defense Data Scientist III   
31                         Defense Data Scientist III   
32  Pharmaceutical/Biotech Market Research Data Sc...   
33                             Data Scientist, Junior   
34                                     Data Scientist   
35   Jr. Machine Learning Data Scientist - Python / R   
36                                     Data Scientist   
37              Tech Excellence Data Engineer, Junior   
38                                     Data Scientist   
39                             Data Scientist/Analyst   
40                        Data Scientist (ALL LEVELS)   
41                                     Data Scientist   
42                                     Data Scientist   
43                              Junior Data Scientist   
44                    Data Scientist / Data Analytics   
45                                     Data Scientist   
46                            Aviation Data Scientist   
47                                     Data Scientist   
48      Data Scientist (Hybrid) (Intelligence Center)   
49                                     Data Scientist   

                                Salary Estimate  \
0                 $76K - $120K (Glassdoor est.)   
1                $101K - $138K (Glassdoor est.)   
2                 $74K - $109K (Glassdoor est.)   
3                $116K - $161K (Glassdoor est.)   
4                 $76K - $108K (Glassdoor est.)   
5         Employer Provided Salary:$65K - $205K   
6                 $87K - $126K (Glassdoor est.)   
7        Employer Provided Salary:$100K - $125K   
8        Employer Provided Salary:$125K - $185K   
9                 $96K - $137K (Glassdoor est.)   
10       Employer Provided Salary:$160K - $200K   
11       Employer Provided Salary:$110K - $115K   
12               Employer Provided Salary:$215K   
13        Employer Provided Salary:$50K - $300K   
14                 $46K - $73K (Glassdoor est.)   
15       Employer Provided Salary:$120K - $130K   
16                $83K - $125K (Glassdoor est.)   
17        Employer Provided Salary:$62K - $117K   
18        Employer Provided Salary:$96K - $237K   
19                $91K - $131K (Glassdoor est.)   
20                 $48K - $74K (Glassdoor est.)   
21            

In [4]:
#drop duplicate job postings 
data_frame.drop_duplicates(inplace = True)
data_frame.shape

(3440, 12)

Number of rows got reduced from 5578 to 3440 after removing the duplictaes

In [5]:
# to check if there are any null values in Salary column
data_frame['Salary Estimate'].isna().sum()

0

In [6]:
# checking for types of values Salary Estimate column holds
data_frame['Salary Estimate'].value_counts()

$135K - $175K (Employer est.)             155
-1                                        134
$65K - $205K (Employer est.)              110
$180K - $260K (Employer est.)              86
$210K - $304K (Employer est.)              76
                                         ... 
Employer Provided Salary:$140K - $145K      1
$77K - $108K (Glassdoor est.)               1
$67K - $97K (Glassdoor est.)                1
$78K - $113K (Glassdoor est.)               1
$112K - $157K (Glassdoor est.)              1
Name: Salary Estimate, Length: 810, dtype: int64

In [7]:
# deleting 134 rows with no salary information (-1 values). This is around 0.04% of the total number of rows
data_frame = data_frame[data_frame['Salary Estimate'] != '-1']
data_frame.shape

(3306, 12)

In [8]:
data_frame['Salary Estimate'].value_counts()

$135K - $175K (Employer est.)             155
$65K - $205K (Employer est.)              110
$180K - $260K (Employer est.)              86
$210K - $304K (Employer est.)              76
$120K - $140K (Employer est.)              67
                                         ... 
Employer Provided Salary:$140K - $145K      1
$77K - $108K (Glassdoor est.)               1
$67K - $97K (Glassdoor est.)                1
$78K - $113K (Glassdoor est.)               1
$112K - $157K (Glassdoor est.)              1
Name: Salary Estimate, Length: 809, dtype: int64

In [9]:
# Creating separate columns for following types of salary 
# 1. Per Hour
# 2. Employer Provided Salary
# 3. Employer est.
data_frame['hourly'] = data_frame['Salary Estimate'].apply(lambda x: 1 if 'per hour' in x.lower() else 0)
data_frame['employer provided'] = data_frame['Salary Estimate'].apply(lambda x: 1 if 'employer provided salary' in x.lower() else 0)
data_frame['employer estimate'] = data_frame['Salary Estimate'].apply(lambda x: 1 if 'employer est.' in x.lower() else 0)

In [10]:
# to check if there are values in the new columns
data_frame.loc[data_frame['hourly']==1]

Job Title  \
23                              Senior Data Scientist   
21                                     Data Scientist   
16                    Data Scientist/Big Data Analyst   
26                                     Data Scientist   
6                           Sr. Research Data Analyst   
20                                     Data Scientist   
12                  Analytics Developer, Data Science   
28                     Athletics Gameday Statistician   
3    Artificial Intelligence Documentation Specialist   
4       Data Scientist(Data Bricks experience)-Hybrid   
3                                   Data Scientist IV   
10                                     Data Scientist   
11                                     Data Scientist   
14                                     Data Scientist   
20                                Senior Data Analyst   
14                Data Scientist /Data Analyst--H4551   
28                                 Game Mathematician   
11           Data Analyst - Senior - Decision Support   
28  Senior Health Care Research / Senior Healthcar...   
22                Basketball Scorekeeper/Statistician   
25                                     Data Scientist   
23                             Data Analyst/Scientist   
25                                  Data Scientist II   
4   Federal Work Study Student - Athletics Statist...   
11                                       Statistician   
13                           IT Data Scientist (NC48)   
16                   Chemical Exposure Data Scientist   
17                                     Data Scientist   
24                   Data Scientist, Machine Learning   
25                                     DATA SCIENTIST   
27                   Chemical Exposure Data Scientist   
33                   Data Scientist, Machine Learning   
42                          Machine Learning Engineer   
3                                      Data Scientist   
11                                       Data Analyst   
12                           Associate Data Scientist   
10                                     Data Scientist   
6                                      Data Scientist   
3                          Digital Data Scientist (H)   
21                            Data Scientist [Remote]   

                                 Salary Estimate  \
23   Employer Provided Salary:$70 - $75 Per Hour   
21   Employer Provided Salary:$75 - $80 Per Hour   
16   Employer Provided Salary:$50 - $60 Per Hour   
26   Employer Provided Salary:$65 - $70 Per Hour   
6    Employer Provided Salary:$28 - $38 Per Hour   
20   Employer Provided Salary:$45 - $65 Per Hour   
12   Employer Provided Salary:$35 - $52 Per Hour   
28         Employer Provided Salary:$25 Per Hour   
3    Employer Provided Salary:$20 - $23 Per Hour   
4    Employer Provided Salary:$35 - $46 Per Hour   
3          Employer Provided Salary:$84 Per Hour   
10   Employer Provided Salary:$70 - $75 Per Hour   
11         Employer Provided Salary:$85 Per Hour   
14   Employer Provided Salary:$70 - $80 Per Hour   
20         Employer Provided Salary:$57 Per Hour   
14   Employer Provided Salary:$40 - $45 Per Hour   
28   Employer Provided Salary:$20 - $59 Per Hour   
11   Employer Provided Salary:$43 - $67 Per Hour   
28   Employer Provided Salary:$33 - $50 Per Hour   
22   Employer Provided Salary:$13 - $15 Per Hour   
25   Employer Provided Salary:$65 - $70 Per Hour   
23         Employer Provided Salary:$40 Per Hour   
25   Employer Provided Salary:$57 - $65 Per Hour   
4          Employer Provided Salary:$13 Per Hour   
11   Employer Provided Salary:$33 - $39 Per Hour   
13   Employer Provided Salary:$58 - $63 Per Hour   
16         Employer Provided Salary:$31 Per Hour   
17   Employer Provided Salary:$70 - $75 Per Hour   
24         Employer Provided Salary:$50 Per Hour   
25   Employer Provided Salary:$63 - $65 Per Hour   
27         Employer Provided Salary:$31 Per Hour   
33         Employer Provided Salary:$63 Pe

We observed all the rows with string 'per hour' also contain string 'employer provided salary'.

In [11]:
# Removing text after bracket from salary column
salary = data_frame['Salary Estimate'].apply(lambda x: x.split('(')[0])

In [12]:
salary.head(40)

0                                   $76K - $120K 
1                                  $101K - $138K 
2                                   $74K - $109K 
3                                  $116K - $161K 
4                                   $76K - $108K 
5           Employer Provided Salary:$65K - $205K
6                                   $87K - $126K 
7          Employer Provided Salary:$100K - $125K
8          Employer Provided Salary:$125K - $185K
9                                   $96K - $137K 
10         Employer Provided Salary:$160K - $200K
11         Employer Provided Salary:$110K - $115K
12                 Employer Provided Salary:$215K
13          Employer Provided Salary:$50K - $300K
14                                   $46K - $73K 
15         Employer Provided Salary:$120K - $130K
16                                  $83K - $125K 
17          Employer Provided Salary:$62K - $117K
18          Employer Provided Salary:$96K - $237K
19                                  $91K - $131K 


In [13]:
# Removing K and $ from the salary column
minus_Kd = salary.apply(lambda x: x.replace('K','').replace('$',''))

In [14]:
# to check how many rows still contains any alphabet values 
minus_Kd.str.contains('[a-zA-Z]').sum()

291

In [15]:
# Number of rows with Employer in them
minus_Kd.str.contains('Employer').sum()

291

All the alphabetic rows have word 'Employer' in them. If we remove all the text from the employer rows, it should work.

In [16]:
min_hr = minus_Kd.apply(lambda x: x.lower().replace('per hour','').replace('employer provided salary:',''))

In [17]:
data_frame['min_salary'] = min_hr.apply(lambda x: int(x.split('-')[0]) if '-' in x else np.NaN)
data_frame['max_salary'] = min_hr.apply(lambda x: int(x.split('-')[1]) if '-' in x else np.NaN)

In [18]:
data_frame['avg_salary'] =(data_frame.min_salary+data_frame.max_salary)/2

In [19]:
#If range not provided, assigned the sigle value salary available
data_frame['avg_salary'][data_frame['avg_salary'].isna()] = min_hr

C:\Users\WM796GU\AppData\Local\Temp\ipykernel_39552\456859982.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_frame['avg_salary'][data_frame['avg_salary'].isna()] = min_hr


In [20]:
data_frame.head(50)

Job Title  \
0             Geospatial Researcher/Data Scientist IV   
1                               Senior Data Scientist   
2   Software Integrator / Data Scientist - Expert ...   
3                                 Lead Data Scientist   
4                                      Data Scientist   
5                                      Data Scientist   
6                           Defense Data Scientist II   
7                                 Senior Data Analyst   
8                                      Data Scientist   
9                     Defense Senior Data Scientist I   
10                                     Data Scientist   
11                                     Data Scientist   
12                 2023771 Data Scientist $215,000.00   
13                              DATA SCIENTIST - USMS   
14                                Junior Data Labeler   
15                                     Data Scientist   
16                     Tech Excellence Data Scientist   
17                                     Data Scientist   
18                              Senior Data Scientist   
19                                     Data Scientist   
20               Jr. Data Scientist (Remote Eligible)   
21                                   Data Scientist I   
22                                     Data Scientist   
23                              Senior Data Scientist   
24               Data Scientist - Electricity Markets   
25                Data Scientist - Clearance Required   
26                      Data Scientist with Polygraph   
27                                     Data Scientist   
28             Senior Data Scientist (Remote/On Site)   
29                                     Data Scientist   
30                         Defense Data Scientist III   
32  Pharmaceutical/Biotech Market Research Data Sc...   
33                             Data Scientist, Junior   
34                                     Data Scientist   
35   Jr. Machine Learning Data Scientist - Python / R   
36                                     Data Scientist   
37              Tech Excellence Data Engineer, Junior   
38                                     Data Scientist   
39                             Data Scientist/Analyst   
40                        Data Scientist (ALL LEVELS)   
41                                     Data Scientist   
42                                     Data Scientist   
43                              Junior Data Scientist   
44                    Data Scientist / Data Analytics   
45                                     Data Scientist   
46                            Aviation Data Scientist   
47                                     Data Scientist   
48      Data Scientist (Hybrid) (Intelligence Center)   
49                                     Data Scientist   
50                                     Data Scientist   

                                Salary Estimate  \
0                 $76K - $120K (Glassdoor est.)   
1                $101K - $138K (Glassdoor est.)   
2                 $74K - $109K (Glassdoor est.)   
3                $116K - $161K (Glassdoor est.)   
4                 $76K - $108K (Glassdoor est.)   
5         Employer Provided Salary:$65K - $205K   
6                 $87K - $126K (Glassdoor est.)   
7        Employer Provided Salary:$100K - $125K   
8        Employer Provided Salary:$125K - $185K   
9                 $96K - $137K (Glassdoor est.)   
10       Employer Provided Salary:$160K - $200K   
11       Employer Provided Salary:$110K - $115K   
12               Employer Provided Salary:$215K   
13        Employer Provided Salary:$50K - $300K   
14                 $46K - $73K (Glassdoor est.)   
15       Employer Provided Salary:$120K - $130K   
16                $83K - $125K (Glassdoor est.)   
17        Employer Provided Salary:$62K - $117K   
18        Employer Provided Salary:$96K - $237K   
19                $91K - $131K (Glassdoor est.)   
20                 $48K - $74K (Glassdoor est.)   
21            

In [21]:
#state of the job location
df['job_state'] = df['Location'].apply(lambda x: x.split(',')[1])

IndexError: list index out of range

In [22]:
# Figuring out reason for the error.
data_frame['Location'][~data_frame['Location'].str.contains(',')].value_counts()

United States    17
Manhattan        17
California       15
Texas             3
St Louis Park     2
Florida           2
Hawaii            1
Pennsylvania      1
Name: Location, dtype: int64

Rows which have above locations do not contain character ',' in them. This is leading to the above error in accessing the second item of the list.

In [23]:
states = data_frame['Location'][~data_frame['Location'].str.contains(',')].value_counts().index

In [24]:
data_frame['job_state'] = data_frame['Location'].apply(lambda x: x if x in states else x.split(',')[1])

In [25]:
data_frame['job_state'].value_counts()

 CA              610
 TX              278
 OH              253
 FL              253
 NC              182
 NY              163
 MO              106
 MD              106
 VA              102
 MA               97
 AZ               90
 MI               86
 CO               83
 IL               82
 WA               82
 UT               80
 OR               80
 GA               76
 MN               76
 PA               63
 IN               59
 TN               54
 DC               47
 QC               36
 NV               34
 KS               24
 NJ               18
 HI               17
United States     17
Manhattan         17
California        15
 DE                6
 Cuyahoga          3
Texas              3
St Louis Park      2
Florida            2
Hawaii             1
Pennsylvania       1
 KY                1
 SC                1
Name: job_state, dtype: int64

In [26]:
# Removing leading zeros
data_frame['job_state'] = data_frame['job_state'].apply(lambda x: x.lstrip(' '))

In [27]:
# Removing Quebec, Canada jobs
data_frame = data_frame[data_frame['job_state']!='QC']

In [28]:
data_frame['job_state'].value_counts()

CA               610
TX               278
FL               253
OH               253
NC               182
NY               163
MD               106
MO               106
VA               102
MA                97
AZ                90
MI                86
CO                83
WA                82
IL                82
OR                80
UT                80
GA                76
MN                76
PA                63
IN                59
TN                54
DC                47
NV                34
KS                24
NJ                18
HI                17
United States     17
Manhattan         17
California        15
DE                 6
Cuyahoga           3
Texas              3
St Louis Park      2
Florida            2
Hawaii             1
Pennsylvania       1
KY                 1
SC                 1
Name: job_state, dtype: int64

In [29]:
data_frame['avg_salary'].describe()

count     3270.0
unique     270.0
top        155.0
freq       161.0
Name: avg_salary, dtype: float64

In [30]:
# Mapping states with 2 letter abbreviation 
data_frame['job_state'][data_frame['job_state']=='Manhattan']='NY'
data_frame['job_state'][data_frame['job_state']=='California']='CA'
data_frame['job_state'][data_frame['job_state']=='Texas']='TX'
data_frame['job_state'][data_frame['job_state']=='St Louis Park']='MN'
data_frame['job_state'][data_frame['job_state']=='Florida']='FL'
data_frame['job_state'][data_frame['job_state']=='Hawaii']='HI'
data_frame['job_state'][data_frame['job_state']=='Pennsylvania']='PA'
data_frame['job_state'][data_frame['job_state']=='Cuyahoga']='OH'

C:\Users\WM796GU\AppData\Local\Temp\ipykernel_39552\3010522983.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_frame['job_state'][data_frame['job_state']=='California']='CA'
C:\Users\WM796GU\AppData\Local\Temp\ipykernel_39552\3010522983.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_frame['job_state'][data_frame['job_state']=='Texas']='TX'
C:\Users\WM796GU\AppData\Local\Temp\ipykernel_39552\3010522983.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.h

In [31]:
data_frame['job_state'].value_counts()

CA               625
TX               281
OH               256
FL               255
NC               182
NY               180
MD               106
MO               106
VA               102
MA                97
AZ                90
MI                86
CO                83
IL                82
WA                82
UT                80
OR                80
MN                78
GA                76
PA                64
IN                59
TN                54
DC                47
NV                34
KS                24
NJ                18
HI                18
United States     17
DE                 6
KY                 1
SC                 1
Name: job_state, dtype: int64

In [32]:
data_frame.head()

Job Title  \
0            Geospatial Researcher/Data Scientist IV   
1                              Senior Data Scientist   
2  Software Integrator / Data Scientist - Expert ...   
3                                Lead Data Scientist   
4                                     Data Scientist   

                  Salary Estimate  \
0   $76K - $120K (Glassdoor est.)   
1  $101K - $138K (Glassdoor est.)   
2   $74K - $109K (Glassdoor est.)   
3  $116K - $161K (Glassdoor est.)   
4   $76K - $108K (Glassdoor est.)   

                                     Job Description  Rating  \
0  Battelle delivers when others can’t. We conduc...     3.6   
1  JDSAT, Inc. is a solutions company - utilizing...     4.3   
2  *** Must have TS/SCI ***\nJOB DESCRIPTION:\nIn...     4.6   
3  Tensure is a fully-remote technology consultin...     3.8   
4  JDSAT, Inc. is a solutions company - utilizing...     4.3   

           Company Name          Location                    Size  Founded  \
0              Battelle  Fort Belvoir, VA  1001 to 5000 Employees     1929   
1                 JDSAT        McLean, VA       1 to 50 Employees       -1   
2  Que Technology Group        McLean, VA       1 to 50 Employees       -1   
3    Tensure Consulting    Washington, DC       1 to 50 Employees       -1   
4                 JDSAT        McLean, VA       1 to 50 Employees       -1   

        Type of ownership                Industry                   Sector  \
0  Nonprofit Organization  Research & Development  Management & Consulting   
1       Company - Private                      -1                       -1   
2       Company - Private                      -1                       -1   
3       Company - Private                      -1                       -1   
4       Company - Private                      -1                       -1   

                      Revenue  hourly  employer provided  employer estimate  \
0     $5 to $10 billion (USD)       0                  0                  0   
1    Unknown / Non-Applicable       0                  0                  0   
2  Less than $1 million (USD)       0                  0                  0   
3    Unknown / Non-Applicable       0                  0                  0   
4    Unknown / Non-Applicable       0                  0                  0   

   min_salary  max_salary avg_salary job_state  
0        76.0       120.0       98.0        VA  
1       101.0       138.0      119.5        VA  
2        74.0       109.0       91.5        VA  
3       116.0       161.0      138.5        DC  
4        76.0       108.0       92.0        VA

In [33]:
data_frame['job_state'].isnull().sum()

0

In [34]:
#age of company 
data_frame['age'] = data_frame.Founded.apply(lambda x: x if x <1 else 2022 - x)

In [35]:
data_frame.head()

Job Title  \
0            Geospatial Researcher/Data Scientist IV   
1                              Senior Data Scientist   
2  Software Integrator / Data Scientist - Expert ...   
3                                Lead Data Scientist   
4                                     Data Scientist   

                  Salary Estimate  \
0   $76K - $120K (Glassdoor est.)   
1  $101K - $138K (Glassdoor est.)   
2   $74K - $109K (Glassdoor est.)   
3  $116K - $161K (Glassdoor est.)   
4   $76K - $108K (Glassdoor est.)   

                                     Job Description  Rating  \
0  Battelle delivers when others can’t. We conduc...     3.6   
1  JDSAT, Inc. is a solutions company - utilizing...     4.3   
2  *** Must have TS/SCI ***\nJOB DESCRIPTION:\nIn...     4.6   
3  Tensure is a fully-remote technology consultin...     3.8   
4  JDSAT, Inc. is a solutions company - utilizing...     4.3   

           Company Name          Location                    Size  Founded  \
0              Battelle  Fort Belvoir, VA  1001 to 5000 Employees     1929   
1                 JDSAT        McLean, VA       1 to 50 Employees       -1   
2  Que Technology Group        McLean, VA       1 to 50 Employees       -1   
3    Tensure Consulting    Washington, DC       1 to 50 Employees       -1   
4                 JDSAT        McLean, VA       1 to 50 Employees       -1   

        Type of ownership                Industry                   Sector  \
0  Nonprofit Organization  Research & Development  Management & Consulting   
1       Company - Private                      -1                       -1   
2       Company - Private                      -1                       -1   
3       Company - Private                      -1                       -1   
4       Company - Private                      -1                       -1   

                      Revenue  hourly  employer provided  employer estimate  \
0     $5 to $10 billion (USD)       0                  0                  0   
1    Unknown / Non-Applicable       0                  0                  0   
2  Less than $1 million (USD)       0                  0                  0   
3    Unknown / Non-Applicable       0                  0                  0   
4    Unknown / Non-Applicable       0                  0                  0   

   min_salary  max_salary avg_salary job_state  age  
0        76.0       120.0       98.0        VA   93  
1       101.0       138.0      119.5        VA   -1  
2        74.0       109.0       91.5        VA   -1  
3       116.0       161.0      138.5        DC   -1  
4        76.0       108.0       92.0        VA   -1

In [36]:
##parsing of job description (python, etc.)
#python
data_frame['python_yn'] = data_frame['Job Description'].apply(lambda x: 1 if 'python' in x.lower() else 0)
data_frame.python_yn.value_counts()

0    2981
1     289
Name: python_yn, dtype: int64

In [37]:
#SQL
data_frame['sql_yn'] = data_frame['Job Description'].apply(lambda x: 1 if 'sql' in x.lower() else 0)
data_frame.sql_yn.value_counts()

0    3078
1     192
Name: sql_yn, dtype: int64

In [39]:
data_frame.to_csv('salary_data_cleaned.csv', index = False)